### DISCUSSION TITLE
I can't understand the lut-baking process in FColorGrading initializing functon

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#7229] [Q&A] [answered]
Hi, I  have see the lut-baking implementation code listed below, so there are some step in lut-baking:
1.do logC encoding, change value fro logc to linear
2.kill negative
3.colorGradingIn depend on the tonemapping choose

Let's say if i choose `ACES_LEGACY` tone mapping, so `colorGradingIn` will move the value in `srgb` color space to `REC2020`. But my question is the value after `LogC_to_linear` isn't linear? Why assume it is in `srgb` color space, i can't understand.

```
  for (size_t g = 0; g < config.lutDimension; g++) {
      for (size_t r = 0; r < config.lutDimension; r++) {
          float3 v = float3{r, g, b} * (1.0f / float(config.lutDimension - 1u));

          // LogC encoding
          v = LogC_to_linear(v);

          // Kill negative values near 0.0f due to imprecision in the log conversion
          v = max(v, 0.0f);

          if (builder->hasAdjustments) {
              // Exposure
              v = adjustExposure(v, builder->exposure);

              // Purkinje shift ("low-light" vision)
              v = scotopicAdaptation(v, builder->nightAdaptation);
          }

          // Move to color grading color space
          v = c.colorGradingIn * v;
     }
  }
               
```

**Comment by dandingol03** [comment]
@romainguy give me some help if it is convinient for you

**Reply by romainguy** [reply]
This has nothing to do with the current color space, it's just compression. Why do you need to know about those details?

**Reply by dandingol03** [reply]
@romainguy  Because our artist meet the chromatism difference between `blender` and `filament`, and this difference is so obvious that we can not ignore it . So i read the color management code in `blender` and `filament`, and try to figure out why there is some chromatism when we both choose 'filmic' tone mappiing. 
So i just need the answer of this question: the `colorGradingIn` function suppose the current colorspace is `srgb`. Does this `srgb`  is the same thing  with `linear rgb`?  Because i always treat `srgb` as the srgb color space which gamma = 0.45, i can't understand how  the colorspace with gamma which is not _1_ can be called `linear`.

**Reply by romainguy** [reply, marked as answer]
`LogC` has nothing to do with the final colors, is just a way to encode the values in the LUT. The opposite is done when reading from the LUT.

Filament's "filmic" curve is not the same as Blender's "filmic" curve, it's not more complicated than that :) I'm planning on adding the AgX tone mapping curve to Filament, which is also available in Blender.

And to answer your question, sRGB is a confusing name. You can have "linear sRGB" which really means the Rec709 gamut with a linear transfer function (and you'll see that in the code we use more precise definitions like this: `ColorSpace outputColorSpace = Rec709-sRGB-D65;`). But again the `LogC_to_linear` function has nothing to do with this.

**Reply by dandingol03** [reply]
Thanks for your kindly reply.

**Reply by romainguy** [reply]
See https://github.com/google/filament/pull/7236

This will match Blender 4.0.
